packages/invoices/src/InboundShipment/DetailView/DetailView.tsx

```diff
@@ -1,60 +1,108 @@
 import React, { FC } from 'react';
 import { useParams } from 'react-router';
 import {
-  Column,
+  useQuery,
   TableProvider,
   createTableStore,
   useOmSupplyApi,
-  useDocument,
   useColumns,
   GenericColumnKey,
   getNotePopoverColumn,
   getRowExpandColumn,
   useDialog,
   DialogButton,
   useTranslation,
-  Item,
+  useMutation,
+  useQueryClient,
 } from '@openmsupply-client/common';
 
-import { InboundAction, reducer } from './reducer';
 import { getInboundShipmentDetailViewApi } from './api';
 import { Toolbar } from './Toolbar';
 import { Footer } from './Footer';
 import { AppBarButtons } from './AppBarButtons';
 import { SidePanel } from './SidePanel';
 import { GeneralTab } from './GeneralTab';
 import { InboundLineEdit } from './modals/InboundLineEdit/InboundLineEdit';
-
-import { isInboundEditable } from '../../utils';
-import { InboundShipmentItem } from '../../types';
+import { inboundLinesToSummaryItems } from './reducer/reducer';
+import {
+  getNextInboundStatus,
+  isInboundEditable,
+  placeholderInbound,
+} from '../../utils';
+import {
+  InboundShipmentItem,
+  InboundShipment,
+  Invoice,
+  OutboundShipmentSummaryItem,
+} from '../../types';
 
 const useDraftInbound = () => {
+  const queryClient = useQueryClient();
```

Comment:
is a bit rushed for now but gives a good example.

Not using the reducer at all here. Which means that sorting isn't working. But it's a lot of code and complexities removed which is good, I think.